pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -726,44 +725,31 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 					if err != nil {
 						return err
 					}
-					// In case Portworx if the snapshot ID is populated for every volume then the snapshot
-					// process is considered to be completed successfully.
-					// This ensures we don't execute the post-exec before all volume's snapshot is completed
-					if driverName == volume.PortworxDriverName {
-						startTime := time.Now()
-						for {
-							// In case below logic genuinely takes long time for many volumes
-							// the CR timestamp need to be updated. This prevents backup timeout error.
-							elapsedTime := time.Since(startTime)
-							if elapsedTime > utils.TimeoutUpdateBackupCrTimestamp {
-								backup, err = a.updateBackupCRInVolumeStage(
-									namespacedName,
-									stork_api.ApplicationBackupStatusInProgress,
-									backup.Status.Stage,
-									"Volume backups are in progress",
-									volumeInfos,
-								)
-								if err != nil {
-									return err
-								}
-								startTime = time.Now()
-							}
-							// Get fresh stock of the status for all volumes.
-							snapshotNotCompleted := false
-							volumeInfos, err = driver.GetBackupStatus(backup)
-							if err != nil {
-								log.ApplicationBackupLog(backup).Errorf("error getting backup status for driver %v: %v", driverName, err)
-								return fmt.Errorf("error getting backup status for driver %v: %v", driverName, err)
-							}
-							for _, volInfo := range volumeInfos {
-								if volInfo.BackupID == "" {
-									log.ApplicationBackupLog(backup).Tracef("Snapshot of volume %v is not done yet, need to loop till snapshot is done...", volInfo.PersistentVolumeClaim)
-									snapshotNotCompleted = true
-								}
-							}
-							if !snapshotNotCompleted {
-								break
-							}
+				}
+			}
+
+			// In case Portworx if the snapshot ID is populated for every volume then the snapshot
+			// process is considered to be completed successfully.
+			// This ensures we don't execute the post-exec before all volume's snapshot is completed
+			for driverName := range pvcMappings {
+				var driver volume.Driver
+				driver, err = volume.Get(driverName)
+				if err != nil {
+					return err
+				}
+				if driverName == volume.PortworxDriverName {
+					volumeInfos, err := driver.GetBackupStatus(backup)
+					if err != nil {
+						return fmt.Errorf("error getting backup status: %v", err)
+					}
+					for _, volInfo := range volumeInfos {
+						if volInfo.BackupID == "" {
+							log.ApplicationBackupLog(backup).Infof("Snapshot of volume [%v] hasn't completed yet, let's retry snapshot", volInfo.PersistentVolumeClaim)
```

Comment:
let's retry snapshot"  is confusing, here we are not going to retry the snapshot, we will retry checking of status again. Please accordingly.